src/main/java/xyz/wagyourtail/jsmacros/client/api/library/impl/FPlayer.java

```diff
@@ -126,36 +124,177 @@ public boolean writeSign(String l1, String l2, String l3, String l4) {
     }
 
     /**
-     * @see #takeScreenshot(String, String, MethodWrapper)
-     *
-     * @since 1.2.6
      * @param folder
      * @param callback calls your method as a {@link Consumer}&lt;{@link TextHelper}&gt;
+     * @see #takeScreenshot(String, String, MethodWrapper)
+     * @since 1.2.6
      */
     public void takeScreenshot(String folder, MethodWrapper<TextHelper, Object, Object> callback) {
         assert folder != null;
         ScreenshotUtils.saveScreenshot(new File(Core.instance.config.macroFolder, folder), mc.getWindow().getFramebufferWidth(), mc.getWindow().getFramebufferHeight(),
-            mc.getFramebuffer(), (text) -> {
-                if (callback != null) callback.accept(new TextHelper(text));
-            });
+                mc.getFramebuffer(), (text) -> {
+                    if (callback != null) callback.accept(new TextHelper(text));
+                });
     }
-    
+
     /**
      * Take a screenshot and save to a file.
-     *
+     * <p>
      * {@code file} is the optional one, typescript doesn't like it not being the last one that's optional
      *
-     * @since 1.2.6
-     * 
      * @param folder
      * @param file
      * @param callback calls your method as a {@link Consumer}&lt;{@link TextHelper}&gt;
+     * @since 1.2.6
      */
     public void takeScreenshot(String folder, String file, MethodWrapper<TextHelper, Object, Object> callback) {
         assert folder != null && file != null;
         ScreenshotUtils.saveScreenshot(new File(Core.instance.config.macroFolder, folder), file, mc.getWindow().getFramebufferWidth(), mc.getWindow().getFramebufferHeight(),
-            mc.getFramebuffer(), (text) -> {
-                if (callback != null) callback.accept(new TextHelper(text));
-            });
+                mc.getFramebuffer(), (text) -> {
+                    if (callback != null) callback.accept(new TextHelper(text));
+                });
+    }
+
+    /**
+     * Creates a new PlayerInput object.
+     *
+     * @return a new {@link xyz.wagyourtail.jsmacros.client.api.classes.PlayerInput PlayerInput}.
+     * @see xyz.wagyourtail.jsmacros.client.api.classes.PlayerInput
+     * @since 1.4.0
+     */
+    public PlayerInput createPlayerInput() {
```

Comment:
create overloads for this (params can only be doubles, cast to float internally) and I think that'll be it